projects/igniteui-angular/src/lib/grid/column.component.ts

```diff
@@ -349,23 +777,36 @@ export class IgxColumnComponent implements AfterContentInit {
             }
         }
     }
+/**
+ * Updates the highlights when a column index is changed.
+ * ```typescript
+ * this.column.updateHighlights(1, 3);
+ * ```
+ * @memberof IgxColumnComponent
+ */
+public updateHighlights(oldIndex: number, newIndex: number) {
+    const activeInfo = IgxTextHighlightDirective.highlightGroupsMap.get(this.grid.id);
+
+    if (activeInfo && activeInfo.columnIndex === oldIndex) {
+        IgxTextHighlightDirective.setActiveHighlight(this.grid.id, {
+            columnIndex: newIndex,
+            rowIndex: activeInfo.rowIndex,
+            index: activeInfo.index,
+            page: activeInfo.page,
+        });
 
-    public updateHighlights(oldIndex: number, newIndex: number) {
-        const activeInfo = IgxTextHighlightDirective.highlightGroupsMap.get(this.grid.id);
-
-        if (activeInfo && activeInfo.columnIndex === oldIndex) {
-            IgxTextHighlightDirective.setActiveHighlight(this.grid.id, {
-                columnIndex: newIndex,
-                rowIndex: activeInfo.rowIndex,
-                index: activeInfo.index,
-                page: activeInfo.page,
-            });
-
-            this.grid.refreshSearch(true);
-        }
+        this.grid.refreshSearch(true);
     }
-
-    public pin(index?) {
+}
+/**
+ * Pins the column.
```

Comment:
Pins the column at the provided index in the pinned area. Defaults to index 0 if not provided.